server.go

```diff
@@ -1473,6 +1512,8 @@ func nextConnID() uint64 {
 const DefaultMaxRequestBodySize = 4 * 1024 * 1024
 
 func (s *Server) serveConn(c net.Conn) error {
+	defer s.wg.Done()
```

Comment:
defer is too heavy. it simplify code, but it costs too much in this case. please, no defer on hot paths.

Reply:
1) `defer` used to be heavy in previous versions of Go where it would heap allocate for the closure. In recent versions of Go (after 1.8 IIRC) the compiler does analysis and stack allocates the space so it has only little overhead left: https://gist.github.com/erikdubbelboer/90ca1b7045edbae9a263b267812af550 (only about 43ns per call)
2) `serveConn` is only called once per connection which means this only adds 43ns of overhead per connection, not per request. I think the 43ns is worth the code complexity of having to add the statement in each return path and having all of fasthttp lock up in case of a panic.